client/landing/stepper/declarative-flow/onboarding.ts

```diff
@@ -344,6 +346,18 @@ const onboarding: Flow = {
 			state: isLoading ? AssertConditionState.CHECKING : AssertConditionState.SUCCESS,
 		};
 	},
+	useSideEffect( currentStepSlug ) {
+		const dispatch = useDispatch();
+		const [ , , clearAllStepPersistedState ] = useStepPersistedState( 'onboarding' );
+
+		useEffect( () => {
+			if ( ! currentStepSlug ) {
+				resetOnboardStore();
+				dispatch( setSelectedSiteId( null ) );
+				clearAllStepPersistedState();
```

Comment:
The onboarding store is already being reset in the `site-setup-flow` [here](https://github.com/Automattic/wp-calypso/blob/6b6d701f8101637b9a5e10f5c3238ae62acff380/client/landing/stepper/declarative-flow/site-setup-flow.ts#L250). I wonder if it make sense to also call the `clearAllStepPersistedState` there so everything is cleaned up after the flow ends

Reply:
I think it's ok to move the logic for clearing the state to the front of the flow. But I guess that means we should have a follow up task that removes the `resetOnboardStore` call from the `site-setup` flow.

Reply:
I got a WIP https://github.com/Automattic/wp-calypso/pull/97972